templates/docker-compose.yml

```diff
@@ -36,6 +36,53 @@ services:
 {% endif %}
     volumes:
       - ./lemmy.hjson:/config/config.hjson:Z
+    command:
+      - "--disable-scheduled-tasks"
+      - "--disable-activity-sending"
+    depends_on:
+      - postgres
+      - pictrs
+
+  lemmy-schedule:
+    image: {{ lemmy_docker_image }}
+    hostname: lemmy-schedule
+    restart: always
+    logging: *default-logging
+    environment:
+{% if lemmy_env_vars is defined and lemmy_env_vars|length > 0 %}
+{% for item in lemmy_env_vars %}
+{% for key, value in item.items() %}
+      - {{ key }}={{ value }}
+{% endfor %}
+{% endfor %}
+{% endif %}
+    volumes:
+      - ./lemmy.hjson:/config/config.hjson:Z
+    command:
+      - "--disable-activity-sending"
+      - "--disable-http-server"
+    depends_on:
+      - postgres
+      - pictrs
+
+  lemmy-federate:
+    image: {{ lemmy_docker_image }}
+    hostname: lemmy-federate
+    restart: always
+    logging: *default-logging
+    environment:
+{% if lemmy_env_vars is defined and lemmy_env_vars|length > 0 %}
+{% for item in lemmy_env_vars %}
+{% for key, value in item.items() %}
+      - {{ key }}={{ value }}
+{% endfor %}
+{% endfor %}
+{% endif %}
+    volumes:
+      - ./lemmy.hjson:/config/config.hjson:Z
+    command:
+      - "--disable-scheduled-tasks"
+      - "--disable-http-server"
```

Comment:
I could go either way on separating these... mostly I'm worried about the increased memory requirements on smaller servers, from running these 3 alongside each other.

Reply:
Would be worth testing if it really makes much difference. Most likely overall memory usage would be similar, as you have the same data stored in memory either in a single process or split across multiple processes. But seeing as the only benefit here is to have separate logs, I would keep it in a single process. That way the docker-compose file is also easier to edit manually (its very easy to forget changing the version in one place).

Reply:
I agree then. 

@ticoombs if you want to show an example of the 3 servers just for documentation/examples sakes, you could move them into a differently named but unused `docker-compose.split_server.yml` or something, but leave this change out of the original file.

Reply:
if versions/image name is the concern this is easily solvable by yaml references, similar to what is already used e.g. in the lemmy woodpecker config: https://github.com/LemmyNet/lemmy/blob/f229f09f92f44bb67fce3801be5345d89a0b808a/.woodpecker.yml#L5

in a compose file you can specify a section with an `x-` prefix for this purpose: https://docs.docker.com/compose/compose-file/11-extension/

if resource usage is a concern it could be gated by a variable that is defaulting to false.

it would likely be beneficial to maintain a dedicated non-jinja compose file as reference for non-ansible-users, as people have already repeatedly been struggling with trying to understand the jinja templated file in here when they were trying to do non-ansible setups.

Reply:
My main concern is about complexity, Id prefer to keep it simple.

> it would likely be beneficial to maintain a dedicated non-jinja compose file as reference for non-ansible-users, as people have already repeatedly been struggling with trying to understand the jinja templated file in here when they were trying to do non-ansible setups.

Yes definitely.